config/routes.rb

```diff
@@ -21,6 +21,8 @@
 
   get '/reset_pin', to: 'reset_pins#index'
   post '/reset_pin', to: 'reset_pins#reset'
+  get '/change_pin/:token', to: 'reset_pins#change_form', as: :change_pin_with_token
```

Comment:
Open to other options here on how we craft a URL with a token in it. Tried to keep it simple.

Reply:
Also, we could possibly implement  (future)`change_pin_with_pin` more of a regular "change my password" type workflow